python/src/com/jetbrains/python/codeInsight/imports/PythonImportUtils.java

```diff
@@ -234,20 +237,31 @@ private static boolean isPossibleModuleReference(PyElement node) {
 
   private static Collection<PsiElement> findImportableModules(PsiFile targetFile, String reftext, Project project, GlobalSearchScope scope) {
     List<PsiElement> result = new ArrayList<>();
-    PsiFile[] files = FilenameIndex.getFilesByName(project, reftext + ".py", scope);
-    for (PsiFile file : files) {
-      if (isImportableModule(targetFile, file)) {
-        result.add(file);
+    //add packages
+    final PsiManager psiMgr = PsiManager.getInstance(project);
+    FileBasedIndex.getInstance().processValues(FilenameIndex.NAME, reftext, null, (file, value) -> {
```

Comment:
There is no API that achieves the desired effect of avoiding a rather expensive operaton of converting/resolving VirtualFile to PsiFile (psiManager.findFile). Those APIs do not provide for that...